arch-design/delfin/RequirementsDesignSpec.md

```diff
@@ -0,0 +1,278 @@
+    
+5. **Removing/Unregistering a Storage Device from Monitoring**
+
+    User invokes an API to DELETE already added storage system. All discovered metadata and other monitoring config will be deleted.
+    
+6. **Real-time monitoring of storage device alarms** 
+
+    User receives alarm information through a configured exporter . When the system is abnormal, SNMP alarm packets are reported.
+    system  Receives and parses alarm messages, and pushes alarm outside through exporters.  
+7. **Query Alarm information from device** 
+
+    User invokes API to get rrent alarms (default) and historical alarms, including serial number, severity, policy name, name, devices, first occurrence time, type, and status.
+    
+    
+8. **Clearing Alarms**
+
+    User select an alarm ID and invokes an API to clear from the system. Delfin clears this alarm from the device management platform.
+    
+9. **Storage Device Performance Information**
+
+    Detailed analysis and models available [here](https://github.com/sodafoundation/architecture-analysis/blob/master/arch-design/delfin/PerfomanceMontoringDesign.md) 
+
+
+
+
+#### Usecase context model
+NA
+
+### Interface Model
+
+
+#### External Interfaces
+REST API is provided for configuring devices for monitoring and for querying collected information.  
+Detailed API spec is available here [here](https://github.com/sodafoundation/delfin/blob/master/openapi-spec/swagger.yaml) 
+
+#### Internal Interfaces
+* Driver Interface : Provides class interfaces for the drivers to implement back end attributes collection. [Driver APIs](https://github.com/sodafoundation/delfin/blob/master/delfin/drivers/api.py)
+
+* Exporter Interface : Provides class interfaces for the external exporters to get data from Delfin to third party platforms.
+[Exporter APIS](https://github.com/sodafoundation/delfin/blob/master/delfin/exporter/base_exporter.py)
+* DB Interface : Provides interfaces to access DB . [DB APIS](https://github.com/sodafoundation/delfin/blob/master/delfin/db/api.py)
+
+### End User Context 
+NA
+
+### Functional Context
+
+NA
+
+### Non Functional Context
+
+#### Performance
+NA
+#### Security
+NA
+#### Other Non Functional Aspects
+NA
+
+### Data View
+
+Detailed data model design 
+[here](https://github.com/sodafoundation/architecture-analysis/blob/28f8f477a1d8ae3ec1793fc8ea677a0518898339/Analysis/DataStorageProjects/Delfin/ResourceModel/ResourceModel.md)
+
+NAS capability model design [here](https://github.com/sodafoundation/architecture-analysis/blob/648dd3c9fad6fbc51f31ffc51ba392ebd9a14859/Analysis/DataStorageProjects/Delfin/NASModels/NASResources.md)
+
+### Development and Deployment Context
+NA
+
+### Execution View
+
+NA
+
+
+
+
+## Sequence Diagrams
+### Storage registration
+![Register](resources/Register.png)
+
+### Resource synchronization 
+
+![Resource sync](resources/Sync.jpg)
+
+### Query resource information 
+
+![Resource sync](resources/Query_workflow.png)
```

Comment:
IMO, Its a broken link as Query_workflow.jpg file is available in "Resources" directory.

Reply:
Done , thanks